gwcs/selector.py

```diff
@@ -1,87 +1,168 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
 from __future__ import division, print_function
 
 import copy
 import numpy as np
 import warnings
 
+from astropy.extern import six
 from astropy.modeling.core import Model
-from astropy.utils.misc import isiterable
-
+from astropy.modeling.parameters import Parameter
 
+from . import region
 from .util import RegionError
 
 
-class SelectorModel(Model):
+def _toindex(value):
+    """
+    Convert value to an int or an int array.
+
+    Input coordinates should be turned into integers
+    corresponding to the center of the pixel.
+    They are used to index the mask.
+
+    Examples
+    --------
+    >>> _toindex(np.array([-0.5, 0.49999]))
+    array([0, 0])
+    >>> _toindex(np.array([0.5, 1.49999]))
+    array([1, 1])
+    >>> _toindex(np.array([1.5, 2.49999]))
+    array([2, 2])
+    """
+    indx = np.empty(value.shape, dtype=np.int32)
+    indx = np.floor(value + 0.5, out=indx)
+    return indx
+
+
+class SelectorMask(Model):
+
     """
+    A mask model to be used with the `~gwcs.selector.RegionsSelector` transform.
+
+    For an IFU observation, the values of the mask
+    correspond to the region (slice)  label.
+
     Parameters
     ----------
-    labels : a list of strings or objects
-    transforms : a list of transforms
-        transforms match labels
+    mask : ndarray
+        An array of integers or strings where the values
+        correspond to a transform label in `~gwcs.selector.RegionsSelector` model.
+        If a transform is not defined the value shoul dbe set to 0 or " ".
     """
-    _param_names = ()
+
+    inputs = ('x', 'y')
+    outputs = ('z')
 
     linear = False
     fittable = False
 
-    def __init__(self, labels, transforms):
-        # labels should be unique
-        if (np.unique(labels)).size != np.asarray(labels).size:
-            raise ValueError("Labels should be unique.")
-        # every label should have a corresponding transform
-        if len(labels) != len(transforms):
-            raise ValueError("Expected labels and transforms to be of equal length.")
-        if " " in labels or 0 in labels:
-            raise ValueError('"0" and " " are not allowed as labels.')
-        self._selector = dict(zip(labels, transforms))
-        super(SelectorModel, self).__init__(n_models=1)
+    def __init__(self, mask):
+        if mask.dtype.type is not np.string_:
+            self._mask = np.asarray(mask, dtype=np.int)
+        else:
+            self._mask = mask
+        if mask.dtype.type is np.string_:
+            self._no_transform_value = ""
+        else:
+            self._no_transform_value = 0
+        super(SelectorMask, self).__init__()
 
```

Comment:
Just to close the loop over here from this comment (https://github.com/spacetelescope/pyasdf/pull/84#issuecomment-88202662):

We're getting this error message

```
ValueError: Inconsistent data column lengths: set([])
```

because `repr(SelectorMask)` doesn't work.  I haven't looked into why, but it's probably because it fails to convert to a `Table`.  Fixing that should get a better error message out of pyasdf.